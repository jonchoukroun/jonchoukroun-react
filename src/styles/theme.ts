export default {
    colors: {
        green: "#008080",
        blue: "#000080",
        black: "#141414",
        lightGray: "#f8f8f8",
        gray: "#c0c0c0",
        darkGray: "#878787",
        white: "#fbfbfb",
    },
    header: {
        fontWeight: "500",
        transform: "scaleX(1.1),",
    },
    body: {
        fontSize: "18px",
    },
    border: {
        line: "2px solid",
        color: "#f8f8f8 #7f787f #7f787f #f8f8f8",
        radius: "0",
    },
    defaultDropShadow: "1px 2px 0px 1px rgba(20, 20, 20, 0.7)",
    darkDropShadow: "1px 1px 0px 3px rgba(0, 0, 0, 0.7)",
};
