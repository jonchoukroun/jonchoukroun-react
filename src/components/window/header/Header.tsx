import React, { FC } from 'react';
import { Row, Title, HeaderButton } from './styles';

import styled from 'styled-components';

interface IHeaderProps {
    title: string;
    shouldMinimize: boolean;
    shouldClose: boolean;
}

const HeaderButtons = styled.div`
    display: flex;
    flex-direction: row;
    height: auto;
    width: auto;
`;

const Header: FC<IHeaderProps> = ({ title, shouldMinimize, shouldClose }) => {
    return (
        <Row>
            <Title>{title}</Title>
            <HeaderButtons>
                {/* TODO replace buttons with icons */}
                {shouldMinimize ? <HeaderButton>-</HeaderButton> : null}
                {shouldClose ? <HeaderButton>&times;</HeaderButton> : null}
            </HeaderButtons>
        </Row>
    );
};

export default Header;
