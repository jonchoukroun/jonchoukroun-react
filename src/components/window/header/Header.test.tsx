import React from "react";
import Header from "./Header";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import theme from "../../../styles/theme";
import { ThemeProvider } from "styled-components";

const title = "Hey nong man";

it("renders successfully", () => {
    shallow(
        <ThemeProvider theme={theme}>
            <Header title={title} shouldMinimize={true} shouldClose={true} />
        </ThemeProvider>
    );
});

it("renders the title and no buttons", () => {
    const component = renderer.create(
        <ThemeProvider theme={theme}>
            <Header title={title} shouldMinimize={false} shouldClose={false} />
        </ThemeProvider>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders the close button", () => {
    const component = renderer.create(
        <ThemeProvider theme={theme}>
            <Header title={title} shouldMinimize={false} shouldClose={true} />
        </ThemeProvider>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders both buttons", () => {
    const component = renderer.create(
        <ThemeProvider theme={theme}>
            <Header title={title} shouldMinimize={true} shouldClose={true} />
        </ThemeProvider>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
