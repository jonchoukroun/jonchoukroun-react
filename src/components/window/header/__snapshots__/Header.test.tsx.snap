// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders both buttons 1`] = `
<div
  className="sc-AxjAm fnUNlU"
>
  <h4
    className="sc-AxirZ bpGOUS"
  >
    Hey nong man
  </h4>
  <div
    className="sc-AxhCb liSLKZ"
  >
    <button
      className="sc-AxiKw hbMXWK"
    >
      -
    </button>
    <button
      className="sc-AxiKw hbMXWK"
    >
      ×
    </button>
  </div>
</div>
`;

exports[`renders the close button 1`] = `
<div
  className="sc-AxjAm fnUNlU"
>
  <h4
    className="sc-AxirZ bpGOUS"
  >
    Hey nong man
  </h4>
  <div
    className="sc-AxhCb liSLKZ"
  >
    <button
      className="sc-AxiKw hbMXWK"
    >
      ×
    </button>
  </div>
</div>
`;

exports[`renders the title and no buttons 1`] = `
<div
  className="sc-AxjAm fnUNlU"
>
  <h4
    className="sc-AxirZ bpGOUS"
  >
    Hey nong man
  </h4>
  <div
    className="sc-AxhCb liSLKZ"
  />
</div>
`;
